akka-stream/src/main/scala/akka/stream/impl/io/TcpConnectionStream.scala

```diff
@@ -278,6 +279,11 @@ private[akka] class OutboundTcpStreamActor(processorPromise: Promise[Processor[B
       processorPromise.success(exposedProcessor)
       initSteps.become(Actor.emptyBehavior)
 
+    case CommandFailed(_, Some(ex)) ⇒
+      localAddressPromise.failure(ex)
+      processorPromise.failure(ex)
+      fail(ex)
```

Comment:
Instead of repeating the implementation, I’d extract the `Option[Throwable]` and use `.getOrElse(new StreamTcpException(...))` to get the exception out.